### ISSUE TITLE
When feeding rotated point clouds as input, the model performance degrades significantly.

### ISSUE THREAD

**Issue by Dominic23331** [manycore-research/SpatialLM#127]
When I use the ScanNet dataset for inference with the SpatialLM model, I find that the accuracy drops significantly for scenes where the walls are not aligned with the coordinate axes. Although random rotation augmentation is enabled in the training configuration, is the model still sensitive to point cloud rotation in its detection performance?

**Comment by SamMaoYS** [comment]
If I understand correctly, you tried finetuning the model on ScanNet with rotation augmentation enabled. I would expect the model to learn to detect object with rotated scenes properly, since I also finetuned it on ARKitScenes with rotation augmentation and the performance seemed reasonable.
A few things that might be worth checking:
- whether the scenes are consistently z-up,
- whether the rotation augmentation is applied only around the z-axis,
- and whether there are isolated noisy or floating points in the ScanNet scenes that could shift the coordinate distribution away from the training distribution.
- and whether the issue could be related to ScanNet using only axis-aligned bounding boxes (AABB). After scene rotation, the original box annotations may no longer remain axis-aligned. You could try updating the box attributes after rotation so that they are converted back to AABB coordinates. This may lead to some degradation in performance in box fitting.

Thank you!

**Comment by Dominic23331** [comment]
> If I understand correctly, you tried finetuning the model on ScanNet with rotation augmentation enabled. I would expect the model to learn to detect object with rotated scenes properly, since I also finetuned it on ARKitScenes with rotation augmentation and the performance seemed reasonable. A few things that might be worth checking:
> 
> * whether the scenes are consistently z-up,
> * whether the rotation augmentation is applied only around the z-axis,
> * and whether there are isolated noisy or floating points in the ScanNet scenes that could shift the coordinate distribution away from the training distribution.
> * and whether the issue could be related to ScanNet using only axis-aligned bounding boxes (AABB). After scene rotation, the original box annotations may no longer remain axis-aligned. You could try updating the box attributes after rotation so that they are converted back to AABB coordinates. This may lead to some degradation in performance in box fitting.
> 
> Thank you!

I only used the pre-trained model provided in the repository for inference on the ScanNet dataset, without performing any fine-tuning. I suspect that the pretrained model may not have enabled random rotation augmentation, or the probability of applying random rotation is too low.

**Comment by SamMaoYS** [comment]
I see, yes, the pretrained model does not have random rotation augmentation enabled.
